### ISSUE TITLE
StyledText increases row height (depending on text)

### ISSUE THREAD

**Issue by tmssngr** [eclipse-platform/eclipse.platform.swt#691]
**Describe the bug**
(this bug is a duplicate of https://bugs.eclipse.org/bugs/show_bug.cgi?id=527917)
When scrolling or typing some text that can't be rendered by the currently set font, `StyledText` will increase the row height. This makes it hard for other controls that show other line based information beside the `StyledText` control at the right line.

**To Reproduce**
Run this snippet:
```java
import java.io.*;
import java.nio.charset.*;

import org.eclipse.swt.*;
import org.eclipse.swt.custom.*;
import org.eclipse.swt.graphics.*;
import org.eclipse.swt.layout.*;
import org.eclipse.swt.widgets.*;

public class Bug527917 {

	public static void main(String[] args) throws IOException {
		if (args.length != 1) {
			return;
		}

		final StringBuilder buffer = new StringBuilder();
		int lineNo = 0;
		try (BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(args[0]), StandardCharsets.UTF_8))) {
			for (String line = reader.readLine(); line != null; line = reader.readLine()) {
				buffer.append(++lineNo);
				buffer.append(" ");
				buffer.append(line);
				buffer.append("\n");
			}
		}

		final Display display = new Display();
		final Font font = new Font(display, "Liberation Mono", 11, 0);

		final Shell shell = new Shell(display);
		shell.setLayout(new FillLayout());

		final StyledText styledText = new StyledText(shell, SWT.BORDER | SWT.READ_ONLY | SWT.V_SCROLL | SWT.H_SCROLL);
		styledText.setText(buffer.toString());
		styledText.setFont(font);

		shell.setSize(1000, 700);
		shell.open();

		while (!shell.isDisposed()) {
			if (!display.readAndDispatch()) {
				display.sleep();
			}
		}

		font.dispose();
		display.dispose();
	}
}
```
by providing the downloaded file from https://bugs.eclipse.org/bugs/attachment.cgi?id=271697 as command line argument.
Look at how many rows are displayed, for my test on Windows 11 ~24.5 lines are shown. Now scroll down the whole file (holding Page down) and back to the top again. Now only ~23 lines are shown, so the row height has been increased.

**Expected behavior**
The characters that can't be rendered by the current font must not change the row height.
Maybe, instead, they should be rendered using the existing row height.

**Screenshots**
Initial display:
<img width="491" alt="initially" src="https://github.com/eclipse-platform/eclipse.platform.swt/assets/6029561/23d61c6f-7c20-4d2f-9ee5-3fd75f1da4f8">
after scrolling down and up again:
<img width="491" alt="after-scrolling-down-and-back-to-top" src="https://github.com/eclipse-platform/eclipse.platform.swt/assets/6029561/7eb6e586-d017-4917-9dae-62334279b37d">

**Environment:**
1. Select the platform(s) on which the behavior is seen:
- - [x] All OS
- - [ ] Windows
- - [ ] Linux
- - [ ] macOS 

This problem largely depends on the set font and the (more or less randomly) picked replacement font.

**Comment by deepika-u** [comment]
@tmssngr 
I am not able to recreate this issue anymore.
I think this is addressed by #932 and its pr #1002 

 Can you cross verify and confirm please?

**Comment by tmssngr** [comment]
Sorry for the delay. Yes, Alex' fix seems to resolve this problem finally.
